subprojects/maven/src/main/java/org/gradle/api/publish/maven/internal/publication/MavenPomInternal.java

```diff
@@ -57,23 +55,12 @@ public interface MavenPomInternal extends MavenPom {
 
     MavenProjectIdentity getProjectIdentity();
 
-    Set<MavenDependency> getApiDependencyManagement();
-
-    Set<MavenDependency> getRuntimeDependencyManagement();
-
-    Set<MavenDependency> getImportDependencyManagement();
-
-    Set<MavenDependencyInternal> getApiDependencies();
-
-    Set<MavenDependencyInternal> getRuntimeDependencies();
-
-    Set<MavenDependencyInternal> getOptionalApiDependencies();
-
-    Set<MavenDependencyInternal> getOptionalRuntimeDependencies();
+    Property<MavenPomDependencies> getDependencies();
```

Comment:
❔ Why use a `Property` here instead of just getting the dependencies?  So far, it seems to only add an extra step of calling `.get()` everywhere?  Do you have future plans for this?

Reply:
The primary advantage of this is that we can lazily derive the dependencies from the parsed component. We don't need to have some kind of listener to call a `setDependencies` on the POM when the parsed component is eventually realized. 

This line would not be possible otherwise: 
https://github.com/gradle/gradle/blob/3a63caab88bd8446523094b3e72fcf9fcd0c3552/subprojects/maven/src/main/java/org/gradle/api/publish/maven/internal/publication/DefaultMavenPublication.java#L196